### ISSUE TITLE
Daily max/min reference temperatures same as daily means in EAM output

### ISSUE THREAD

**Issue by mabrunke** [E3SM-Project/E3SM#2817]
At the All-Hands Meeting, I presented my finding that the EAM output in at least the AMIP run includes daily maximum and minimum 2-m reference temperatures that are the same as the daily means.  It was discussed that this should be fixed for v2.  In components/cam/src/physics/cam/cam_diagnostics.F90, TREFHTMN and TREFHTMX are given a definition of M (minimum) and X (maximum) to addfld, but it doesn't look like there is any constraint for the minimum/maximum values to be included in these output variables.

**Comment by golaz** [comment]
@wlin7 : do you think this is a bug in the code, or did we do something wrong when we configured the output?

**Comment by wlin7** [comment]
The default flags for TREFHTMX and TREFHTMN were overridden by avgflag_pertape ('A') for the daily fincl2 as used for the DECKv1 simulations.  Specs of 'TREFHTMX:X','TREFHTMN:M' in daily tape would work regardless of avgflag_pertage. 

Also thanks to @swrneale for the offline comments on explicit specification of flags for TREFHTMX and TREFHTMN

**Comment by golaz** [comment]
@swrneale and @wlin7 : thanks for helping out with this.

Here is the `fincl2` line in the run e3sm script that was used for piControl:

https://github.com/E3SM-Project/SimulationScripts/blob/df5197c1fd3002343f4c3912afbb8329afc31363/archive/DECKv1/ne30_oEC/run_acme.20180129.DECKv1b_piControl.ne30_oEC.edison.csh#L1007

If I understand you properly, we should replace

`fincl2 = [...] ,'TREFHT','TREFHTMN','TREFHTMX', [...]`

with

`fincl2 = [...] ,'TREFHT','TREFHTMN:M','TREFHTMX:X', [...]`

**Comment by wlin7** [comment]
Yes, @golaz , that will work with the updated fincl2, despite the 'A' in avgflag_pertape.

**Comment by golaz** [comment]
@wlin7, thanks for confirming. On a slightly related note; is it possible to directly output the **monthly mean of daily min/max**? I assume if we add

`'TREFHTMN:M','TREFHTMX:X'`

to fincl1, we will just get the **monthly min/max** values, which is not what it typically compared to observations.

**Comment by wlin7** [comment]
@golaz , you are right. If we put 'TREFHTMN:M','TREFHTMX:X' on fincl1, we would get monthly min/max values as the "M' and 'X' flag is intended for -- the min/max of the output period, instead of monthly mean of daily min/max. I don't see an easy way of directly saving such monthly means given the current history file module.
